app/controllers/FileValidationController.scala

```diff
@@ -59,8 +60,12 @@ class FileValidationController @Inject()(
               updatedAnswers <- Future.fromTry(UserAnswers(request.internalId).set(ValidXMLPage, fileName))
               updatedAnswersWithURL <- Future.fromTry(updatedAnswers.set(URLPage, downloadUrl))
               _              <- sessionRepository.set(updatedAnswersWithURL)
+              xml <- Future.successful(xmlValidationService.loadXML(downloadUrl))
```

Comment:
you don't have to wrap stuff up in a future for the for, you can set it equal ie xml = xmlValid...